aot-methods.txt

```diff
@@ -0,0 +1,351 @@
+java/io/DataOutputStream.close.()V
+javax/microedition/lcdui/CanvasLFImpl.uCallKeyPressed.(I)V
+javax/microedition/lcdui/Canvas.keyPressed.(I)V
+com/sun/midp/events/NativeEvent.<init>.(I)V
+com/ibm/oti/connection/file/Connection.openOutputStream.(J)Ljava/io/OutputStream;
+com/ibm/oti/connection/file/FCOutputStream.<init>.([BJLcom/ibm/oti/connection/file/Connection;)V
+com/sun/midp/rms/RecordStoreImpl.<init>.(Lcom/sun/midp/security/SecurityToken;ILjava/lang/String;Z)V
+java/lang/Character.toUpperCase.(C)C
+com/sun/midp/events/NativeEvent.<init>.()V
+java/util/Hashtable.clear.()V
+com/sun/cldc/util/j2me/CalendarImpl.floorDivide.(II[I)I
+java/io/ByteArrayInputStream.close.()V
+com/sun/cldc/i18n/Helper.normalizeEncodingName.(Ljava/lang/String;)Ljava/lang/String;
+com/sun/midp/midletsuite/MIDletSuiteImpl.getProperty.(Ljava/lang/String;)Ljava/lang/String;
+com/sun/midp/midletsuite/SuiteProperties.getProperty.(Ljava/lang/String;)Ljava/lang/String;
+com/sun/midp/util/Properties.getProperty.(Ljava/lang/String;)Ljava/lang/String;
+com/ibm/oti/connection/file/Connection.checkWrite.()V
+javax/microedition/lcdui/DisplayableLFImpl.lGetDisplayable.()Ljavax/microedition/lcdui/Displayable;
+com/sun/midp/security/SecurityInitializerImpl.<init>.(Lcom/sun/midp/security/SecurityToken;[Ljava/lang/String;)V
+com/sun/midp/midletsuite/SuiteProperties.loadProperties.()V
+com/sun/midp/rms/RecordStoreIndex.getRecordIDs.()[I
+java/io/DataInputStream.readLong.()J
+com/sun/midp/main/MIDletProxyList.notifyListenersOfProxyUpdate.(Lcom/sun/midp/main/MIDletProxy;I)V
+com/nokia/mid/s40/codec/DataEncoder.<init>.(Ljava/lang/String;)V
+java/lang/Exception.<init>.(Ljava/lang/String;)V
+java/lang/Throwable.<init>.(Ljava/lang/String;)V
+java/util/Vector.removeElement.(Ljava/lang/Object;)Z
+com/ibm/oti/connection/file/FCOutputStream.close.()V
+com/ibm/oti/connection/file/Connection.notifyOutputStreamClosed.()V
+javax/microedition/lcdui/Display.getCurrent.()Ljavax/microedition/lcdui/Displayable;
+javax/microedition/lcdui/Image.createImage.(Ljava/lang/String;)Ljavax/microedition/lcdui/Image;
+javax/microedition/lcdui/ImageDataFactory.createResourceImageData.(Ljava/lang/String;)Ljavax/microedition/lcdui/ImageData;
+javax/microedition/lcdui/ImageData.<init>.()V
+javax/microedition/lcdui/ImageDataFactory.loadAndCreateImmutableImageDataFromCache.(Ljavax/microedition/lcdui/ImageData;Ljava/lang/String;)Z
+javax/microedition/lcdui/SuiteImageCacheImpl.loadAndCreateImmutableImageData.(Ljavax/microedition/lcdui/ImageData;Ljava/lang/String;)Z
+com/sun/midp/io/Util.toJavaString.([B)Ljava/lang/String;
+java/util/Hashtable$HashtableEnumerator.nextElement.()Ljava/lang/Object;
+com/sun/midp/lcdui/DisplayDevice.isPrimaryDisplay.()Z
+com/sun/midp/main/MIDletProxy.getClassName.()Ljava/lang/String;
+com/sun/midp/rms/RmsEnvironment.getSecureFilenameBase.(I)Ljava/lang/String;
+com/sun/midp/midletsuite/SuiteContainerAdapter.getSecureFilenameBase.(I)Ljava/lang/String;
```

Comment:
There are 351 methods in this list, but only 349 get compiled on my machine, although "0 failed":

> Compiling ...
> js build/jsc.js -cp java/classes.jar -d -jf java/classes.jar -mff aot-methods.txt > build/classes.jar.js
> Compiled 349 methods OK, 0 failed.

One of the two missing methods is presumably the one I mentioned earlier. It isn't clear what the other one is. But the compiler should warn when a method is missing, so we can prune the list accordingly.